Content.Server/Body/Systems/MetabolizerSystem.cs

```diff
@@ -171,7 +171,7 @@ private void TryMetabolize(EntityUid uid, MetabolizerComponent meta, OrganCompon
                     // still remove reagents
                     if (EntityManager.TryGetComponent<MobStateComponent>(solutionEntityUid.Value, out var state))
                     {
-                        if (_mobStateSystem.IsDead(solutionEntityUid.Value, state))
+                         if (_mobStateSystem.IsDead(solutionEntityUid.Value, state) && !proto.WorksOnTheDead)
```

Comment:
```suggestion
                        if (!proto.WorksOnTheDead && _mobStateSystem.IsDead(solutionEntityUid.Value, state))
```

Flipping the order of these is a slight optimization because of [short circuit evaluation](https://www.geeksforgeeks.org/short-circuit-evaluation-in-programming/).